### DISCUSSION TITLE
Wallet sign in for games

### DISCUSSION THREAD

**Discussion by Nighty13** [cardano-foundation/cardano-wallet#4045] [Show and tell]
### The problem that you wish to solve

Hello ,
currently im trying to push cardano integration for games as much as possible & see how far i can go .
so the idea is the player download Daedalus , launch the game & he will be able to see & connect to his Wallets ( locally game will connect to cardano-wallet & use same Daedalus DB)... all cool to this point , you can sign transactions & do a lot of stuff..
but the moment i tried to implement a sign in , i got stuck .. 

![image](https://user-images.githubusercontent.com/44137382/196009174-744ba429-9f26-4724-ae7b-c7883d9141cf.png)

NOTE : this is Desktop C++ App ... icant use web based wallets etc ...( as many suggested to me ) .
### Description

is there a way to use cardano-wallet to do "message signing" & "verification" (just like today browser wallets do)? so i can verify that the player is truely the owner of x wallet without firing a transaction ..

**Comment by teddyjfpender** [comment]
Hi @Nighty13 - thanks for this!

I agree, this would a message signing feature is something we should support in cardano-wallet. Do you also require a means of verifying a signed message in the wallet? 😄

**Comment by Nighty13** [comment]
Indeed both sign and verify ,  currently i looked into libraries built on top of cardano-wallet ( like cardano-wallet-js and others ) they do handle the message signing themselves everytime .. . having such a feature built into cardano-wallet will be a life saver.

**Comment by Nighty13** [comment]
Hello , i wanted to share with you some work in progress for the unreal engine plugin im building around cardano-wallet api :
https://www.youtube.com/watch?v=CiY5r_Foeoo
